src/Contrib/Zipkin/SpanConverter.php

```diff
@@ -131,4 +145,127 @@ private static function toSpanKind(SpanDataInterface $span): ?int
         
         return null;
     }
+
+    private static function toAnnotation(EventInterface $event): array
+    {
+        $eventName = $event->getName();
+        $attributesAsJson = SpanConverter::convertEventAttributesToJson($event);
+
+        $value = ($attributesAsJson !== null) ? "\"{$eventName}\": {$attributesAsJson}" : "\"{$eventName}\"";
+
+        $annotation = [
+            'timestamp' => AbstractClock::nanosToMicro($event->getEpochNanos()),
+            'value' => $value,
+        ];
+
+        return $annotation;
+    }
+
+    private static function convertEventAttributesToJson(EventInterface $event): ?string
+    {
+        if (count($event->getAttributes()) === 0) {
+            return null;
+        }
+        
+        $attributesArray = [];
+        foreach ($event->getAttributes() as $attr) {
+            $attributesArray[$attr->getKey()] = $attr->getValue();
+        }
+        
+        $attributesAsJson = json_encode($attributesArray);
+        if (($attributesAsJson === false) || (strlen($attributesAsJson) === 0)) {
+            return null;
+        }
+
+        return $attributesAsJson;
+    }
+
+    private static function toRemoteEndpoint(SpanDataInterface $span): ?array
+    {
+        $preferredAttr = SpanConverter::findRemoteEndpointPreferredAttribute($span);
+
+        if ($preferredAttr === null) {
+            return null;
+        }
+
+        if (!is_string($preferredAttr->getValue())) {
+            return null;
+        }
+
+        switch ($preferredAttr->getKey()) {
+            case 'net.peer.ip':
```

Comment:
Could you create/use a constant for this, please?

Reply:
Yup, done